### PR TITLE
Add script to clean ACR looking at GitHub releases

### PR DESCRIPTION
**Pull request by robvanuden**

### CODE REVIEW

**Comment by robvanuden** [comment]
Build in DevOps:
https://dev.azure.com/avivasolutions-public/sitecore-docker/_build?definitionId=3

File: build/Build.RetentionPolicy.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+partial class Build : NukeBuild
+{
+    [Parameter("GitHub repository name to look for in the retention policy")]
+    readonly string GitHubRepositoryName = "avivasolutionsnl/sitecore-docker";
```
Comment (joostmeijles) [inline build/Build.RetentionPolicy.cs]: do not add avivasolutions settings to the repo; use a variable in Azure Pipelines

File: build/Build.RetentionPolicy.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    [Parameter("Azure Container Registry instance name")]
+    readonly string ACRName = "avivasolutionsnl";
+       
+    Target ExecuteRetentionPolicy => _ => _
+        .Executes(async () => {
+            var xcRepositoryNames = XcNames
```
Comment (joostmeijles) [inline build/Build.RetentionPolicy.cs]: please scope this in the file where these are defined e.g. `Build.Xc.cs`

File: execute-retention-policy.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+pr: none
+
+pool:
+  name: Hosted VS2017
+steps:
+- task: AzureCLI@1
+  displayName: 'Execute retention policy'
+  inputs:
+    azureSubscription: 'aviva-sitecore-docker'
```
Comment (joostmeijles) [inline execute-retention-policy.yml]: Use variable

File: execute-retention-policy.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+trigger:
```
Comment (joostmeijles) [inline execute-retention-policy.yml]: Add a comment that this is to be periodically triggered.

File: build/Build.RetentionPolicy.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        foreach (var tag in tagsToBeDeleted)
+        {
+            Console.WriteLine($"Deleting {tag}");
+            ProcessTasks.StartProcess("az", $"acr repository delete -n {registryName} --image {repositoryName}:{tag} --yes").AssertZeroExitCode();
+        }
+    }
+
+    private IEnumerable<string> GetTagsInACR(string registryName, string repositoryName)
+    {
+        var showTagsProcess = ProcessTasks.StartProcess("az", $"acr repository show-tags -n {registryName} --repository {repositoryName}").AssertZeroExitCode();
```
Comment (joostmeijles) [inline build/Build.RetentionPolicy.cs]: Shouldn't you use `Tool` here such that you get automatic error handling (and more Nuke alike syntax)?

Comment (robvanuden) [reply build/Build.RetentionPolicy.cs]: Yes good idea, I changed it

**Pull request merged**